app/controllers/entry_controller.rb

```diff
@@ -44,33 +42,25 @@ def create
   end
 
   def destroy
-    challenge = Challenge.find(params[:challenge])
-    entry = challenge.entries.find(params[:entry])
-
-    if challenge.owner?(current_user) || entry.owner?(current_user)
-      entry.destroy
-      challenge.save
-
-      flash[:notice] = "Deleted entry"
+    if @entry && (@challenge.owner?(current_user) || @entry.owner?(current_user))
```

Comment:
Yeah, think that the rescue in the `load_entry` will redirect if their is an error with `find` and as the `redirect_to` is outside the `if` branch we dont need an `else`